src/btree/bt_handle.c

```diff
@@ -909,15 +909,15 @@ __btree_page_sizes(WT_SESSION_IMPL *session)
     /*
      * Get the allocation size. Allocation sizes must be a power-of-two, nothing else makes sense.
      */
-    WT_RET(__wt_direct_io_size_check(session, cfg, "allocation_size", &btree->allocsize));
+    WT_RET(__wt_allocation_size_check(session, cfg, "allocation_size", &btree->allocsize));
     if (!__wt_ispo2(btree->allocsize))
         WT_RET_MSG(session, EINVAL, "the allocation size must be a power of two");
 
     /*
      * Get the internal/leaf page sizes. All page sizes must be in units of the allocation size.
      */
-    WT_RET(__wt_direct_io_size_check(session, cfg, "internal_page_max", &btree->maxintlpage));
-    WT_RET(__wt_direct_io_size_check(session, cfg, "leaf_page_max", &btree->maxleafpage));
+    WT_RET(__wt_allocation_size_check(session, cfg, "internal_page_max", &btree->maxintlpage));
```

Comment:
Question: Is this the right size check? On new line 912 this is used to check that the value is a power of two. Here on new line 919 the same function is used to check that the value is a multiple of the value on new line 912. This is a different check so shouldn't it have a different function?

Reply:
This is a good question, without direct_io I guess it is no longer needed for the size to units of the allocation size, since this was a direct_io requirement. I'll remove this comment as well as adjust the function name.

Reply:
Updated here [a98162b](https://github.com/wiredtiger/wiredtiger/pull/10430/commits/a98162bc3cb1aa0cffb47e3175091a48c7551a8b). I changed the function name to _get as the function is simply just returning the size now. I have left the checks as I will need to make sure that these checks happening outside the function are no longer required.